### PR TITLE
Add colmap

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
Fix https://github.com/conda-forge/staged-recipes/issues/8899

**Comment by Tobias-Fischer** [comment]
I disabled the Win build because of a weird error that apparently happens in boost 1.74: `%PREFIX%\Library\include\boost/heap/detail/heap_node.hpp(145): error C2300: 'boost::heap::detail::marked_heap_node<std::pair<int,int>>': class does not have a destructor called '~Node'` (fixed in https://github.com/boostorg/heap/pull/29/files)

**Comment by Tobias-Fischer** [comment]
@conda-forge/help-c-cpp this is ready for review.

A few notes:
 - I disabled win for now because of a boost issue (https://github.com/conda-forge/staged-recipes/pull/15744#issuecomment-887179562)
 - The package pulls in a few static libs that aren't in conda-forge. I packaged the licenses for those
 - I did not enable cuda for now - locally I can see that there are errors and I can't be bothered to fix them. I'll create an issue in the feedstock once this PR is merged in.

File: recipes/colmap/build.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+mkdir build
+cd build
+
+cmake ${CMAKE_ARGS} \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DBOOST_STATIC=OFF \
+      -DCUDA_ENABLED=OFF \
+      -DBUILD_SHARED_LIBS=ON \
+      -DCMAKE_INSTALL_LIBDIR=lib \
+      -DCMAKE_INSTALL_PREFIX=$PREFIX \
```
Comment (xhochy) [inline recipes/colmap/build.sh:12]: Part of `CMAKE_ARGS`
```suggestion
```

File: recipes/colmap/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  commands:
+    - if exist %PREFIX%\\Library\\bin\\colmap.exe (exit 0) else (exit 1)  # [win]
+    - test -f $PREFIX/bin/colmap  # [unix]
+    - test -f $PREFIX/lib/colmap/libcolmap.a  # [unix]
+    - test -f $PREFIX/share/colmap/COLMAPConfig.cmake  # [unix]
+
+about:
+  home: https://colmap.github.io/
+  license:  BSD-3-Clause
+  license_file: COPYING.txt
```
Comment (xhochy) [inline recipes/colmap/meta.yaml:94]: You should list all the licenses here
```suggestion
  license_file:
   - COPYING.txt
   - …
```

**Comment by Tobias-Fischer** [comment]
Hi @xhochy, many thanks for the review! I incorporated your requested changes, it would be great if you could have another look :).

**Pull request merged**